### PR TITLE
fix: handling fragment error

### PR DESCRIPTION
**Pull request by nightnei**

### CODE REVIEW

File: lib/request-fragment.js [hunk 1]
```diff
@@ -57,8 +57,12 @@ module.exports = filterHeaders => (
             fragmentRequest.setTimeout(timeout, fragmentRequest.abort);
         }
         fragmentRequest.on('response', response => {
-            if (response.statusCode >= 500) {
-                reject(new Error('Internal Server Error'));
+            if (response.statusCode < 200 || response.statusCode >= 300) {
```
Comment (StyleT) [inline lib/request-fragment.js]: This change behaviour should be in place for non-primary fragments only..

File: lib/request-fragment.js [hunk 2]
```diff
@@ -57,7 +57,12 @@ module.exports = filterHeaders => (
...
         }
         fragmentRequest.on('response', response => {
-            if (response.statusCode >= 500) {
+            const isError500 = response.statusCode >= 500;
+            const isNonPrimaryAndNon200 =
+                (response.statusCode < 200 || response.statusCode >= 300) &&
+                !fragmentAttributes.primary;
+
+            if (isError500 || isNonPrimaryAndNon200) {
                 reject(new Error('Internal Server Error'));
```
Comment (StyleT) [inline lib/request-fragment.js]: Could you PLEASE do a better job at error reporting? With this code we won't be able to see which kind of error we really had... Would be great if you could report at least response status code & body (if available).

Comment (nightnei) [reply lib/request-fragment.js]: I know and I did it previously(https://github.com/StyleT/tailorx/blob/a90ad6ea82b72795d4ce5bcaa3dd95689ba91cfd/lib/request-fragment.js#L63) but decided to revert...

ok, I will do.

Comment (StyleT) [reply lib/request-fragment.js]: Also keep in mind that `Internal Server Error!` text in message is relevant only for 5XX errors.

File: tests/tailor.js [hunk 3]
```diff
@@ -240,6 +240,40 @@ describe('Tailor', () => {
...
+            mockTemplate.returns(
+                '<fragment src="https://fragment/1" primary></fragment>' +
+                    '<fragment src="https://fragment/2" primary></fragment>' +
+                    '<fragment src="https://fragment/3"></fragment>' +
+                    '<fragment src="https://fragment/4"></fragment>'
+            );
+
+            getResponse('http://localhost:8080/test')
+                .then(response => {
+                    console.log(stripComments(response.body));
```
Comment (oleh-momot) [inline tests/tailor.js]: Is this `console.log` needed here?

File: lib/request-fragment.js [hunk 4]
```diff
@@ -57,8 +57,17 @@ module.exports = filterHeaders => (
...
-                reject(new Error('Internal Server Error'));
+            const isError500 = response.statusCode >= 500;
+            const isNonPrimaryAndNon200 =
+                (response.statusCode < 200 || response.statusCode >= 300) &&
+                !fragmentAttributes.primary;
+
+            if (isError500 || isNonPrimaryAndNon200) {
+                reject(
+                    new Error(
+                        `Request fragment error. statusCode: ${response.statusCode}; statusMessage: ${response.statusMessage}; url: ${fragmentUrl};`
```
Comment (oleh-momot) [inline lib/request-fragment.js]: Probably, would be better to use JSON here
What is your opinion about it?

File: tests/tailor.js [hunk 5]
```diff
@@ -240,6 +240,40 @@ describe('Tailor', () => {
...
+                .get('/2')
+                .reply(401, 'hello 2')
+                .get('/3')
+                .reply(300, 'hello 3')
+                .get('/4')
+                .reply(401, 'hello 4');
+
+            mockTemplate.returns(
+                '<fragment src="https://fragment/1" primary></fragment>' +
+                    '<fragment src="https://fragment/2" primary></fragment>' +
```
Comment (oleh-momot) [inline tests/tailor.js]: May we have several primary fragments?

Comment (nightnei) [reply tests/tailor.js]: tailor supports it.
in this case, we will use the first primary fragment information

File: lib/request-fragment.js (hunk 4 above)
Comment (nightnei) [reply lib/request-fragment.js]: what do you mean?

**Pull request merged**